lib/chart/line/buffer.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Chart.Line.Buffer do
+  @moduledoc false
+
+  alias Chart.Internal.AxisLine.MajorTicksText
+  alias Chart.Chart
+
+  def put(%Chart{data: buffer, settings: %{storage_count: count}} = chart, data) do
+    merged_data = merge(buffer, data, count)
+    fl_data = List.flatten(merged_data)
+
+    range_for_x = fl_data |> get_axis_range({1, 0}) |> add_offset_to_range()
+    range_for_y = fl_data |> get_axis_range({0, 1}) |> add_offset_to_range()
+
+    updated_settings =
+      chart.settings
+      |> MajorTicksText.set_range({1, 0}, range_for_x)
+      |> MajorTicksText.set_range({0, 1}, range_for_y)
+
+    chart
+    |> Chart.put_settings(updated_settings)
+    |> Chart.put_data(merged_data)
+  end
+
+  def set_count(settings, count) when is_integer(count) and 0 < count do
+    Map.put(settings, :storage_count, count)
+  end
+
+  # Private
+
+  defp merge(nil, data, _count) do
+    data
+  end
+
+  defp merge([first | _rest] = buffer, data, count) when length(first) <= count do
+    buffer
+    |> Enum.zip(data)
+    |> Enum.map(fn {bf, d} -> Enum.concat(bf, d) end)
+  end
+
+  defp merge(buffer, data, count) do
+    buffer
+    |> Enum.map(fn [_first | rest] -> rest end)
+    |> merge(data, count)
+  end
```

Comment:
Tady netusim, co se deje.

Reply:
Tady ten module by se mel asi prejmenovat na `Stack` (last in first out data structure). Davalo by to lepsi smysl. Tj. pres ten `count`  se nastavi pocet zobrazenych dat. Pak pri kazdem dalsim prichodu dat se prvni odebere a nove prichozi pridaji nakonec...

Reply:
Kdezto `Buffer` by mel jenom ty data kolektovat, tj. ten list by se neustale zvetsoval..